### PR TITLE
Add StreamStats command to stream protocol

### LINKED ISSUE #5412: Add StreamInfo command to stream protocol
Several users ask for a way to get information on a stream, like the first and last available offsets. Requests for other pieces of information, like the number of messages in a stream could come up in the future.

We agreed to add a new `StreamInfo` command for this, it would return a map for extensibility.

```
StreamInfoRequest => Key Version CorrelationId Stream
  Key => uint16 // 0x001c
  Version => uint16
  CorrelationId => uint32
  Stream => string

StreamInfoResponse => Key Version CorrelationId ResponseCode Info
  Key => uint16 // 0x801c
  Version => uint16
  CorrelationId => uint32
  ResponseCode => uint16
  Info => [PieceOfInfo]
  PieceOfInfo => Key Value
  Key => string
  Value => string
```

### PR DESCRIPTION
**Pull request by acogoluegnes**
It returns statistics on a stream, the first and committed chunk IDs for now.

Fixes #5412

### CODE REVIEW

File: deps/rabbitmq_stream/src/rabbit_stream_reader.erl [hunk 1]
```diff
@@ -2641,6 +2640,63 @@ handle_frame_post_auth(Transport,
...
+                        {stream_info, ?RESPONSE_CODE_STREAM_NOT_AVAILABLE, #{}};
+                    {error, not_found} ->
+                        rabbit_global_counters:increase_protocol_counter(stream,
+                                                                         ?STREAM_DOES_NOT_EXIST,
+                                                                         1),
+                        {stream_info, ?RESPONSE_CODE_STREAM_DOES_NOT_EXIST,
+                         #{}};
+                    {ok, MemberPid} ->
+                        OffsetInfo =
+                            case gen:call(MemberPid, '$gen_call',
```
[the remark this answers is not in the document]
Comment (kjnilsson) [reply deps/rabbitmq_stream/src/rabbit_stream_reader.erl:2673]: I have prepared a PR for a new osiris API that can be used for this and that addresses the first offset init issue: https://github.com/rabbitmq/osiris/pull/101

**Review by kjnilsson — CHANGES_REQUESTED** [review]

File: deps/rabbitmq_stream/src/rabbit_stream_reader.erl [hunk 2]
```diff
@@ -2641,6 +2640,63 @@ handle_frame_post_auth(Transport,
...
+                                          get_reader_context)
+                            of
+                                {ok, #{offset_ref := undefined}} ->
+                                    #{};
+                                {ok, #{offset_ref := OffsetRef}} ->
+                                    #{<<"first_offset">> =>
+                                          rabbit_data_coercion:to_binary(
```
[the remark this answers is not in the document]
Comment (acogoluegnes) [reply deps/rabbitmq_stream/src/rabbit_stream_reader.erl]: I switched to `atom_to_binary/1` but `rabbit_data_coercion` in other places of the plugin, I can change the other occurrences in another PR. What's the deal with depending on `rabbit_common`?

File: deps/rabbitmq_stream/docs/PROTOCOL.adoc [hunk 3]
```diff
@@ -702,6 +707,26 @@ CommandVersionsExchangeResponse => Key Version CorrelationId ResponseCode [Comma
...
+
+StreamInfoResponse => Key Version CorrelationId ResponseCode Info
+  Key => uint16 // 0x801c
+  Version => uint16
+  CorrelationId => uint32
+  ResponseCode => uint16
+  Info => [PieceOfInfo]
+  PieceOfInfo => Key Value
+  Key => string
+  Value => string
```
[the remark this answers is not in the document]
Comment (acogoluegnes) [reply deps/rabbitmq_stream/docs/PROTOCOL.adoc]: Using string => int64 instead.

File: deps/rabbitmq_stream/src/rabbit_stream_reader.erl (hunk 1 above)
[the remark this answers is not in the document]
Comment (acogoluegnes) [reply deps/rabbitmq_stream/src/rabbit_stream_reader.erl]: Using new `osiris:get_stats/1` function.

File: deps/rabbitmq_stream/src/rabbit_stream_reader.erl [hunk 4]
```diff
@@ -2641,6 +2640,63 @@ handle_frame_post_auth(Transport,
...
+                         #{}};
+                    {ok, MemberPid} ->
+                        OffsetInfo =
+                            case gen:call(MemberPid, '$gen_call',
+                                          get_reader_context)
+                            of
+                                {ok, #{offset_ref := undefined}} ->
+                                    #{};
+                                {ok, #{offset_ref := OffsetRef}} ->
+                                    #{<<"first_offset">> =>
```
[the remark this answers is not in the document]
Comment (acogoluegnes) [reply deps/rabbitmq_stream/src/rabbit_stream_reader.erl]: Fixed in https://github.com/rabbitmq/osiris/pull/101.

**Review by kjnilsson — APPROVED** [review]

**Pull request merged**